### DISCUSSION TITLE
how to convert json to object with "new" keyword

### DISCUSSION THREAD

**Discussion by Hanbunhanabi** [nlohmann/json#2565] [Q&A] [answered]
Hi. 
this example on the github shows how to convert json to struct:

> // create a person
  ns::person p {"Ned Flanders", "744 Evergreen Terrace", 60};
 // conversion: person -> json
  json j = p;
   std::cout << j << std::endl;
  // {"address":"744 Evergreen Terrace","age":60,"name":"Ned Flanders"}
   // conversion: json -> person
  auto p2 = j.get<ns::person>();
   // that's it
  assert(p == p2);

and my question is: how to convert the json to struct/ Object with "new" keyword : i.e;( ` Person*p3 = new Person()`), and something  like :
`
auto* p2 = new j.get<ns::person>();
`

thank you

**Comment by nlohmann** [comment, marked as answer]
This works:

```cpp
#include <iostream>
#include "json.hpp"

using json = nlohmann::json;

namespace ns {
struct person {
    std::string name;
    std::string address;
    int age;
};

void to_json(json& j, const person& p) {
    j = json{{"name", p.name}, {"address", p.address}, {"age", p.age}};
}

void from_json(const json& j, person& p) {
    j.at("name").get_to(p.name);
    j.at("address").get_to(p.address);
    j.at("age").get_to(p.age);
}
}

int main()
{
    json j = R"({"address":"744 Evergreen Terrace","age":60,"name":"Ned Flanders"})"_json;
    auto* p = new ns::person(j);
    std::cout << p->name << ", " << p->address << ", " << p->age << std::endl;
}
```

**Reply by nlohmann** [reply]
@Hanbunhanabi Do you need further assistance?

**Reply by Hanbunhanabi** [reply]
thank you, this solved my question
